### PR TITLE
Do not show deprecation stack traces in integration tests by default

### PR DESCRIPTION
**Pull request by lptr**
This makes it easier to work with tests that emit deprecation warnings.

### CODE REVIEW

**Comment by lptr** [comment]
@bot-gradle test QFL please.

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/GradleExecuter.java [hunk 1]
```diff
@@ -198,9 +198,9 @@ public void connect(PipedInputStream snk) throws IOException {
     GradleExecuter withNoExplicitNativeServicesDir();
 
     /**
-     * Disables the rendering of stack traces for deprecation logging.
+     * Enables the rendering of stack traces for deprecation logging.
      */
-    GradleExecuter withFullDeprecationStackTraceDisabled();
+    GradleExecuter withFullDeprecationStackTraceEnabled();
```
Comment (wolfs) [inline subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/GradleExecuter.java:203]: Did you try out if a test fails if there is a deprecation warning, even if the deprecation stack traces are disabled?

**Review by wolfs — COMMENTED** [review]
The changes look good so far. There are some CI failures.

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/GradleExecuter.java (hunk 1 above)
Comment (lptr) [reply subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/GradleExecuter.java:203]: I worked with that code, it checks each line if it contains the word "deprecated", it doesn't require the stack trace.

Comment (lptr) [reply subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/GradleExecuter.java:203]: All hail undetected merge conflicts :/

**Comment by lptr** [comment]
@bot-gradle test QFL please.

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/GradleExecuter.java (hunk 1 above)
Comment (lptr) [reply subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/GradleExecuter.java:203]: Merge conflict fixed.

Note that we used to have tests that disabled stack traces in the past, and those worked fine. I only switched over the behavior's default.

**Pull request merged**